### PR TITLE
Refactor Company Class to Client Class

### LINKED ISSUE #159: Refactor Company to Clients
Tasks to do:
- Rename methods
- Update test cases

### PR DESCRIPTION
**Pull request by jicsontoh**

### CODE REVIEW

**Review by ElginL — APPROVED** [review]
LGTM!

File: LICENSE [hunk 1]
```diff
@@ -2,7 +2,7 @@ MIT License
 
 Copyright (c) 2016 Software Engineering Education - FOSS Resources
 
-Permission is hereby granted, free of charge, to any company obtaining a copy
+Permission is hereby granted, free of charge, to any client obtaining a copy
```
Comment (notle1706) [inline LICENSE]: Should this be company instead?

File: docs/UserGuide.md [hunk 2]
```diff
@@ -48,9 +48,9 @@ interactions recorded faster and simpler than traditional GUI applications.
 
    * **`list`** : Lists all companies.
 
-   * **`create`**`create 2 n/James p/97814456 e/test@test.com t/MainPoc t/owesMoney` : Creates a point-of-contact named `James` linked to the company at index 2.
+   * **`create`**`create 2 n/James p/97814456 e/test@test.com t/MainPoc t/owesMoney` : Creates a point-of-contact named `James` linked to the client at index 2.
 
-   * **`delete`**`delete 1` : Deletes the company at the first index together with all their contacts and transactions in the current list.
+   * **`delete`**`delete 1` : Deletes the client at the first index together with all their contacts and transactions in the current list.
 
    * **`find`**`MacDonalds` : Finds MacDonalds in the list of companies, and displays all its details (point-of-contact, transactions).
```
Comment (notle1706) [inline docs/UserGuide.md]: Perhaps `companies` should be changed to clients?

File: docs/DeveloperGuide.md [hunk 3]
```diff
@@ -278,30 +278,30 @@ Priorities: High (must have) - `* * *`, Medium (nice to have) - `* *`, Low (unli
 
 | Priority | As a …​                  | I want to …​                                        | So that I can…​                                                    |
 |----------|--------------------------|-----------------------------------------------------|--------------------------------------------------------------------|
-| `* * *`  | business owner           | choose which company and point-of-contact to delete | save only relevant companies and point-of-contact to remove clutter |
+| `* * *`  | business owner           | choose which client and point-of-contact to delete | save only relevant companies and point-of-contact to remove clutter |
```
Comment (notle1706) [inline docs/DeveloperGuide.md]: Perhapse `companies` should be changed to clients? (This is recurring throughout the document)

**Review by notle1706 — CHANGES_REQUESTED** [review]
Instances of companies should be refactored to clients

File: docs/DeveloperGuide.md (hunk 3 above)
Comment (jicsontoh) [reply docs/DeveloperGuide.md]: okay thanks for pointing out!

File: docs/UserGuide.md (hunk 2 above)
Comment (jicsontoh) [reply docs/UserGuide.md]: Okay sure thank you

File: LICENSE (hunk 1 above)
Comment (jicsontoh) [reply LICENSE]: Sure, will update thank you

**Review by notle1706 — APPROVED** [review]
LGTM

**Pull request merged**